### PR TITLE
fix: Handle missing PageId in metadata when page is resumed

### PR DESCRIPTION
**Pull request by ps863**
This change contains a patch based on customer reported issue, wherein, pageId was missing from the metadata of certain events. 

In a previous [change](https://github.com/aws-observability/aws-rum-web/pull/341) we changed the behavior of when a Page View event is recorded. In cases where  page views resume (this includes resumed sessions and page being reloaded) but page ID is same were changed from being recorded as a separate page view event for each reload/resume to only creating a new page view event only when there was a change in the pageId. However, the change failed to address the following: 

1) In the case when the PageManager's state is restored after being resumed (using the page object stored in cookie), the cookie does not contain PageManager managed attributes which are added as metadata to all events. This means, the PageManager's attributes are not updated from what they were previously and remain null. Therefore, after a page is restored or reloaded, subsequent events don't contain the pageId in the metadata (since pageId is obtained from the PageManager). The pageId is restored in metadata only after a new page view event is generated as that forces the PageManager to update the metadata attributes. 

This change addresses this by: 
1) Updating the PageManager's attributes when a page is resumed as well. 
2) Adding Unit tests to capture this behavior

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sessions/PageManager.ts [hunk 1]
```diff
@@ -108,6 +108,13 @@ export class PageManager {
         } else if (this.page.pageId !== pageId) {
             this.createNextPage(this.page, pageId);
         } else {
```
Comment (adebayor123) [inline src/sessions/PageManager.ts:110]: suggestion: `else if (this.resumed)` might be cleaner

File: src/sessions/__tests__/PageManager.test.ts [hunk 2]
```diff
@@ -233,6 +233,54 @@ describe('PageManager tests', () => {
...
+            interaction: 1
+        });
+
+        window.removeEventListener(
+            'popstate',
+            (pageManager as any).popstateListener
+        );
+    });
+
+    test('when session resumes and page is manually recorded with custom page attributes then custom page attributes are restored', async () => {
```
Comment (adebayor123) [inline src/sessions/__tests__/PageManager.test.ts:248]: Let's add a test case where the pagemanager resumes session with page as null - that should be the scenario we need to capture

File: src/sessions/PageManager.ts [hunk 3]
```diff
@@ -108,6 +108,13 @@ export class PageManager {
...
+            // If its a resumed or reloaded page, update attributes state in PageManager for event metadata
+            if (this.resumed) {
+                this.collectAttributes(
+                    this.page as Page,
+                    typeof payload === 'object' ? payload : undefined
+                );
+            }
```
Comment (qhanam) [inline src/sessions/PageManager.ts:117]:
```suggestion
        } else if (this.resumed) {
            // Update attributes state in PageManager for event metadata
            this.collectAttributes(
                this.page as Page,
                typeof payload === 'object' ? payload : undefined
            );
            return;
        } else {
```

**Review by adebayor123 — APPROVED** [review]

File: src/sessions/PageManager.ts (hunk 3 above)
Comment (adebayor123) [reply src/sessions/PageManager.ts:117]: Approved, but I agree with this change ;)

**Review by qhanam — APPROVED** [review]
